### PR TITLE
Symfony 2.8|^3.0 Support

### PR DESCRIPTION
**Pull request by hmert**
All credits go to @artscorestudio many thanks 🎉

### CODE REVIEW

**Comment by enumag** [comment]
@hmert Any progress? Is there a todo-list what needs to be done?

**Comment by llellouche** [comment]
You can do it !

We wait...

**Comment by llellouche** [comment]
Yes thank you

**Comment by hmert** [comment]
@APY/collaborators hey, we're ready to sf3 :)

File: composer.json [hunk 1]
```diff
@@ -20,9 +20,9 @@
         }
     ],
     "require": {
-        "php": ">=5.4",
-        "symfony/symfony": "~2.3",
-        "twig/twig": ">=1.23.0"
+        "php": ">=5.3.2",
```
Comment (DonCallisto) [inline composer.json]: why `5.3.2`? I would recommend to change this to `5.6`

Comment (qferr) [inline composer.json]: @DonCallisto I agree.

The class name resolution is used in this PR. But this feature is not supported in PHP 5.3.2. 
The PHP version should be `>=5.5`

Comment (qferr) [inline composer.json]: @DonCallisto Yes, indeed. I agree about your recommendation, we should use `5.6`.

Comment (hmert) [inline composer.json]: done

Comment (DonCallisto) [inline composer.json]: @hmert you should also adjust travis: no sense to run in 5.5 env

Comment (hmert) [inline composer.json]: done

**Comment by hmert** [comment]
@APY/collaborators We may merge this branch. Our project has ready to run with this version (`3.0.0` ?). https://travis-ci.org/ojs/ojs/builds/184614260

**Comment by DonCallisto** [comment]
What version of the bundle are we going to choose/release? Because I've introduced months ago a list of callbacks instead of a single one, and we have deprecated something for 2.4
Take a look here https://github.com/APY/APYDataGridBundle/pull/879/files

**Comment by hmert** [comment]
oh great. Let's release it as `2.4`.

**Comment by DonCallisto** [comment]
@hmert so we need to remove code and deprecation. I'm not sure this is a goos idea because we haven't release a tagged version with this deprecation: users may not be aware about this.
If no plans for 3.x version, maybe 2.3 is more suitable.
Otherwise let's icrease the deprecation to 3.x version.

What do all collaborators think?

**Comment by hmert** [comment]
@DonCallisto  let's tag, current master as `2.3.0` and then merge sf3 branch and then release 2.3.1 remove deprecations (and maybe bug fixes) and release 2.4?

**Comment by DonCallisto** [comment]
Well, semantic versioning would suggest us to release a 3.0 version as this is BC (who use symfony 2.3 that was minimum sf version accepted have to upgrade so this seems a BC to me, but I'm not sure).
What I'm prettt sure is this should not be a "patch" version.
Moreover, thinking about semantic versioning, each change made to our public API (like this one) should produce a major version change, so maybe, if you think this is a BC, let's release 3.0 and move deprecation to 4.0 (or remove it directly), if this is a minor for you (that could be the case as I'm not practical with other dependancy bump) let's release 2.4 and modify deprecation text for 3.0 version.

**Comment by DonCallisto** [comment]
Thinking it carefully I think that 2.4 is the best choose possible; so change deprecation message to 3.0, group every other BC due to test, refactor and so on.
Let's see if someone disagree.

**Comment by hmert** [comment]
Ok Here is the plan,
1. Now, tag current master as `2.3.0` before BC, Lot's of changes..  https://github.com/APY/APYDataGridBundle/compare/2.2.17...master
2. People will see the deprecation messages.
3. Release `2.4.0` after cleared deprecations..
4. Merge symfony3 and Release `3.0.0`
5. write tests for sf 2.8 (LTS) and 3.4 (LTS).
6. Develop a sample/test application.

What do you think? @APY/collaborators

**Comment by plfort** [comment]
Nice plan

**Comment by DonCallisto** [comment]
@hmert all good except that `2.4.0`, if necessary, should stay with deprecations: deprecations should be removed into new major (`3.0`)

File: Grid/Grid.php [hunk 2]
```diff
@@ -336,18 +336,18 @@ public function initialize()
...
         $routeParameters = $config->getRouteParameters();
         if (!empty($routeParameters)) {
             foreach ($routeParameters as $parameter => $value) {
                 $this->setRouteParameter($parameter, $value);
             }
         }
+        
+        // Route
+        if (null != $config->getRoute()) {
+            $this->setRouteUrl($this->router->generate($config->getRoute()));
```
Comment (J-Mose) [inline Grid/Grid.php]: It seems that a parameter is missing from the original PR 
was `$this->setRouteUrl($this->router->generate($config->getRoute(), $routeParameters));`

Comment (hmert) [inline Grid/Grid.php]: thanks

**Pull request merged**